carbonmark-api/test/routes/listings/[id]/get.test.ts

```diff
@@ -0,0 +1,49 @@
+import { FastifyInstance } from "fastify";
+import { fixtures } from "../../../fixtures";
+import { build } from "../../../helper";
+import { DEV_URL } from "../../../test.constants";
+import { mockGraphResponses } from "./get.test.mocks";
+
+describe("GET /listings/:id", () => {
+  let fastify: FastifyInstance;
+
+  // Setup the server
+  beforeEach(async () => {
+    try {
+      fastify = await build();
+    } catch (e) {
+      console.error("/listings/:id get.test.ts setup failed", e);
+    }
+  });
+  test("Returns listing with token symbol", async () => {
+    mockGraphResponses({
+      listing: fixtures.marketplace.listing,
+      token: fixtures.tokens.token,
+    });
+    const response = await fastify.inject({
+      method: "GET",
+      url: `${DEV_URL}/listings/0x123`,
+    });
+    expect(response.statusCode).toEqual(200);
+    const json = await response.json();
+    expect(json.symbol).toEqual(fixtures.tokens.token.symbol);
+  });
+
+  test("Returns 404 when listing is not found", async () => {
+    mockGraphResponses({ token: fixtures.tokens.token });
+    const response = await fastify.inject({
+      method: "GET",
+      url: `${DEV_URL}/listings/0x123`,
+    });
+    expect(response.statusCode).toEqual(404);
+  });
+
+  test("Returns 500 when token is not found", async () => {
+    mockGraphResponses({ listing: fixtures.marketplace.listing });
+    const response = await fastify.inject({
+      method: "GET",
+      url: `${DEV_URL}/listings/0x123`,
+    });
+    expect(response.statusCode).toEqual(500);
+  }, 100_000);
```

Comment:
What does the 100_000 do?

Reply:
I bumped the timeout to 100_000 so that jest doesn't kill the process while I debug. Forgot to remove it 😄

Reply:
removed in [9f75ada](https://github.com/KlimaDAO/klimadao/pull/2046/commits/9f75adaae6ff6abb575aa19aa9532d89c4fcd4b4)